go/parquet/internal/utils/bit_writer.go

```diff
@@ -81,11 +81,15 @@ func NewBitWriter(w io.WriterAt) *BitWriter {
 // ReserveBytes reserves the next aligned nbytes, skipping them and returning
 // the offset to use with WriteAt to write to those reserved bytes. Used for
 // RLE encoding to fill in the indicators after encoding.
-func (b *BitWriter) ReserveBytes(nbytes int) int {
+func (b *BitWriter) ReserveBytes(nbytes int) (int, error) {
 	b.Flush(true)
 	ret := b.byteoffset
 	b.byteoffset += nbytes
-	return ret
+	if _, err := b.wr.WriteAt(b.raw[:nbytes], int64(ret)); err != nil {
+		log.Println(err)
+		return 0, err
+	}
```

Comment:
Instead of doing this would it make more sense to expand the `io.WriterAt` interface we're using to be something more like 

```go
type writer interface {
    io.WriterAt
    Reserve(nbytes int)
}
```

so that the `ReserveBytes` method here could just be 

```go
b.Flush(true)
ret := b.byteoffset
b.byteoffset += nbytes
b.wr.Reserve(nbytes)
return ret
```

Thoughts?

Reply:
This looks ok, but I guess it need to modify some code. Let me do it later.

Reply:
@zeroshade I guess a `Reserve` is not enough, maybe it should be "Reserve + Advance"

Reply:
I did a quick update which solves the issue. what do you think? I agree we should probably change the name. Perhaps instead of `Reserve` we can call it `Advance` or `Skip` etc.?

Reply:
Skip LGTM! Thanks!